lib/r10k/source/hash.rb

```diff
@@ -135,16 +135,15 @@ def initialize(name, basedir, options = {})
     @environments_hash = options.delete(:environments) || {}
 
     @environments_hash.keys.each do |name|
-      # TODO: deal with names that aren't valid
-      ::R10K::Util::SymbolizeKeys.symbolize_keys!(@environments_hash[name])
+      R10K::Util::SymbolizeKeys.symbolize_keys!(@environments_hash[name])
       @environments_hash[name][:basedir] = basedir
-      @environments_hash[name][:dirname] = name
+      @environments_hash[name][:dirname] = R10K::Environment::Name.new(name, {prefix: @prefix, source: @name}).dirname
     end
   end
 
   def environments
-    @environments ||= @environments_hash.map do |name, hash|
-      R10K::Environment.from_hash(name, hash)
+    @environments ||= @environments_hash.map do |key, hash|
```

Comment:
why change this from `name`?

Reply:
Good question; there is no good reason. Leftover crumb from previous experiments. Cleaning that up now...